addons/binding/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/handler/BoschIndegoHandler.java

```diff
@@ -0,0 +1,252 @@
+import de.zazaz.iot.bosch.indego.IndegoController;
+import de.zazaz.iot.bosch.indego.IndegoException;
+
+/**
+ * The {@link BoschIndegoHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jonas Fleck - Initial contribution
+ */
+public class BoschIndegoHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(BoschIndegoHandler.class);
+    private int commandToSend;
+
+    private Thread pollingThread;
+    private boolean running;
+
+    public BoschIndegoHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            // Currently manual refrshing is not possible in the moment
+            return;
+        } else if (channelUID.getId().equals(STATE) && command instanceof DecimalType) {
+            if (command instanceof DecimalType) {
+                synchronized (this) {
+                    commandToSend = ((DecimalType) command).intValue();
+                    this.notifyAll();
+                }
+            } else {
+            }
+
+            // Note: if communication with thing fails for some reason,
+            // indicate that by setting the status with detail information
+            // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+            // "Could not control device at IP address x.x.x.x");
+        }
+    }
+
+    @Override
+    public void updateConfiguration(Configuration configuration) {
+        // TODO Auto-generated method stub
+        super.updateConfiguration(configuration);
+
+    }
+
+    private synchronized void poll() throws InterruptedException {
+        // Create controller instance
+        try {
+            IndegoController controller = new IndegoController(getConfig().get("username").toString(),
+                    getConfig().get("password").toString());
+            // Connect to server
+            controller.connect();
+            // Query the device state
+            DeviceStateInformation state = controller.getState();
+            DeviceStatus statusWithMessage = DeviceStatus.decodeStatusCode(state.getState());
+            int eshStatus = getEshStatusFromCommand(statusWithMessage.getAssociatedCommand());
+            int mowed = state.getMowed();
+            int error = state.getError();
+            int statecode = state.getState();
+            boolean ready = isReadyToMow(state.getState(), state.getError());
+
+            if (verifyCommand(commandToSend, eshStatus, state.getState(), error)) {
+                logger.debug("Sending command...");
+                updateState(TEXTUAL_STATE, new StringType("Refreshing..."));
+                controller.sendCommand(getCommandFromEshStatus(commandToSend));
+                commandToSend = 0;
+                try {
+                    for (int i = 0; i < 30 && !Thread.interrupted(); i++) {
```

Comment:
I am not sure I am getting the intention here. If the state does not change, this loop blocks for 30 seconds before leaving the method again? Is this meant to throttle the number of commands sent to the server?